### PR TITLE
Additional validations for proposed transactions. #664

### PR DESCRIPTION
**Pull request by afalaleev**

### CODE REVIEW

File: libraries/chain/proposal_evaluator.cpp [hunk 1]
```diff
@@ -56,17 +56,33 @@ namespace golos { namespace chain {
...
             "Can't combine operations required posting authority and active or owner authority");
         required_total.insert(required_posting.begin(), required_posting.end());
 
+        for (const auto& account: required_total) {
+            FC_ASSERT(
+                nullptr != db().find_account(account),
+                "Account '${account}' for proposed operation doesn't exist", ("account", account));
+        }
+
+        FC_ASSERT(!required_total.empty(), "No operations require approvals");
```
Comment (zxcat) [inline libraries/chain/proposal_evaluator.cpp]: required_total.count() is simpler

File: libraries/chain/proposal_evaluator.cpp [hunk 2]
```diff
@@ -56,17 +56,33 @@ namespace golos { namespace chain {
...
         for (const auto& op : o.proposed_operations) {
             trx.operations.push_back(op.op);
         }
-        trx.validate();
+        trx.set_expiration(db().head_block_time() + STEEMIT_MAX_TIME_UNTIL_EXPIRATION);
+
+        const uint32_t skip_steps =
+                db().skip_authority_check |
+                db().skip_transaction_signatures |
+                db().skip_tapos_check |
```
Comment (zxcat) [inline libraries/chain/proposal_evaluator.cpp]: Reading this code, it's not obvious, what happens inside `db()`. Rewrite this to `golos::chain::database::skip_xxx`, which clearly shows that no `db()` side-effects used.

File: libraries/chain/proposal_evaluator.cpp [hunk 3]
```diff
@@ -122,7 +138,7 @@ namespace golos { namespace chain {
 
         auto check_duplicate = [&](const auto& to_add, const auto& dst) {
             for (const auto& a: to_add) {
-                FC_ASSERT(dst.find(a) == dst.end(), "Can't add already exist approval", ("id", a));
+                FC_ASSERT(dst.find(a) == dst.end(), "Can't add already exist approval '${id}", ("id", a));
```
Comment (zxcat) [inline libraries/chain/proposal_evaluator.cpp]: missing closing `'` in string

File: libraries/chain/proposal_evaluator.cpp (hunk 1 above)
Comment (afalaleev) [reply libraries/chain/proposal_evaluator.cpp]: Fixed

File: libraries/chain/proposal_evaluator.cpp (hunk 2 above)
Comment (afalaleev) [reply libraries/chain/proposal_evaluator.cpp]: Fixed

File: libraries/chain/proposal_evaluator.cpp (hunk 3 above)
Comment (afalaleev) [reply libraries/chain/proposal_evaluator.cpp]: Fixed.

**Pull request merged**